### PR TITLE
Wdfn 118 disable last year checkbox

### PR DESCRIPTION
**Pull request by ayan-usgs**

### CODE REVIEW

File: assets/src/scripts/components/hydrograph/index.js [hunk 1]
```diff
@@ -269,8 +269,31 @@ const plotSROnlyTable = function (elem, {tsKey, variable, methods, visible, data
...
+            .dispatch('change'); // trigger a change event
+    } else {
+        checkbox.property('disabled', null);
+    }
+};
+
+
 const timeSeriesGraph = function (elem) {
-    elem.append('div')
+    elem.call(link(controlLastYearSelect, createStructuredSelector({
```
Comment (ayan-usgs) [inline assets/src/scripts/components/hydrograph/index.js]: Opps.

Comment (ayan-usgs) [reply assets/src/scripts/components/hydrograph/index.js]: Wait... I just had a brain-fart. I did want to do that.

File: assets/src/scripts/components/hydrograph/index.js [hunk 2]
```diff
@@ -269,8 +269,31 @@ const plotSROnlyTable = function (elem, {tsKey, variable, methods, visible, data
...
+    }
+    const variableParmCd = variable.variableCode.value;
+    const availableVariableTimeseries = availableTimeseries.filter(x => x[0] === variableParmCd); // an array of arrays
+    const variableTimeseries = availableVariableTimeseries.map(x => x.slice(1));
+    const compareTimeseriesCountVals = variableTimeseries.map(x => x.map(y => y.compareTimeseriesCount));
+    if (compareTimeseriesCountVals.reduce((a, b) => Number(a) + Number(b), 0) === 0) {
+        checkbox
+            .property('disabled', true)
+            .property('checked', false)
+            .dispatch('change'); // trigger a change event
```
Comment (ayan-usgs) [inline assets/src/scripts/components/hydrograph/index.js]: I'm on the fence about whether or not this is confusing behavior.... :thinking:

File: assets/src/scripts/components/hydrograph/index.js [hunk 3]
```diff
@@ -269,8 +269,31 @@ const plotSROnlyTable = function (elem, {tsKey, variable, methods, visible, data
     }
 };
 
+const controlLastYearSelect = function(elem, {variable, availableTimeseries}) {
+    let checkbox = elem.select('.hydrograph-last-year-input');
+    if (variable === null || !availableTimeseries.length) {
```
Comment (mbucknell) [inline assets/src/scripts/components/hydrograph/index.js]: Wouldn't the control be disabled if the time series length was zero?

File: assets/src/scripts/components/hydrograph/index.js [hunk 4]
```diff
@@ -269,8 +269,31 @@ const plotSROnlyTable = function (elem, {tsKey, variable, methods, visible, data
...
+    } else {
+        checkbox.property('disabled', null);
+    }
+};
+
+
 const timeSeriesGraph = function (elem) {
-    elem.append('div')
+    elem.call(link(controlLastYearSelect, createStructuredSelector({
+        variable: currentVariableSelector,
```
Comment (mbucknell) [inline assets/src/scripts/components/hydrograph/index.js]: So this is going to conflict with the work I'm doing now. We will no longer be showing the availablility of the time series in the parameters list, so I had changed the selector to no longer gather the counts.. I wonder if we could just make this simpler, by using the currentVariableTimeSeriesSelector('compare') (in the timeseries module) and seeing if the length of the timeseries is zero?

Comment (ayan-usgs) [reply assets/src/scripts/components/hydrograph/index.js]: That would work. I'll make that change.

**Pull request merged**